mlflow/sagemaker/container/__init__.py

```diff
@@ -57,10 +61,18 @@ def _serve():
 
     Read the MLmodel config, initialize the Conda environment if needed and start python server.
     """
-    m = Model.load("/opt/ml/model/MLmodel")
-    if pyfunc.FLAVOR_NAME not in m.flavors:
-        raise Exception("Only supports pyfunc models and this is not one.")
-    conf = m.flavors[pyfunc.FLAVOR_NAME]
+    model_config_path = os.path.join(MODEL_PATH, "MLmodel")
+    m = Model.load(model_config_path)
+    if mleap.FLAVOR_NAME in m.flavors and _container_includes_mlflow_source():
```

Comment:
I wonder if we should have a way to specify the flavor. It can be confusing to user that for some models we run python and for some other java.